### ISSUE TITLE
Missing pyupgrade errors (UP006, UP035)

### ISSUE THREAD

**Issue by rademacher-p** [astral-sh/ruff#8698]
Hi, I seem to be missing pyupgrade errors for Python 3.11 and Ruff 0.1.5. Reproduce w/ `ruff --isolated --select UP file.py` on a fresh `venv`. This is all from the CLI, no `pyproject.toml`.
```
# file.py
from typing import Type

foo = Type[list]
```

I ❤️ `ruff`, thanks in advance for any help!

**Comment by charliermarsh** [comment]
Thank you for the kind words!

I think this is a result of https://github.com/astral-sh/ruff/issues/3215 -- we stopped rewriting annotations that are used at runtime (roughly, on the right-hand side of an expression) because it can change behavior. https://github.com/astral-sh/ruff/issues/4843 has some more information. I think we actually _can_ rewrite these in some but not all scenarios. In other words, our current approach is probably too conservative.

**Comment by NeilGirdhar** [comment]
I think you should write this as a type alias:
```
foo: TypeAlias = Type[list]
```
And then would it be possible for Ruff to make the transformation?

**Comment by rademacher-p** [comment]
Aside from the issues with autofix, is there a reason not to provide the deprecation inspection?

**Comment by charliermarsh** [comment]
The only reason is to ensure we're not introducing false positives. So it'd be nice to make sure we have a sense for where this is and isn't safe prior to enabling the check for runtime annotations.

**Comment by charliermarsh** [comment]
Oh hmm, I'm actually noticing that if your Python version is set correctly, we _do_ flag and fix this: https://play.ruff.rs/526e6c64-b8da-4cd5-8dcd-68ee71a6f886

Do you have your `--target-version` configured to Python 3.11?

**Comment by rademacher-p** [comment]
@charliermarsh No, I did not! Should have had that in my config (actually, should just up my `project.requires-python`) 😮 Adding this option gives me the pyupgrade errors - feel free to close, if you're ready 👍

**Comment by charliermarsh** [comment]
Awesome, thanks @rademacher-p! Sorry for the confusion but glad to resolve.
